lib/dialects/oracledb/transaction.js

```diff
@@ -54,21 +54,12 @@ module.exports = class Oracle_Transaction extends Transaction {
   acquireConnection(config, cb) {
     const configConnection = config && config.connection;
     const t = this;
-    return new Bluebird((resolve, reject) => {
-      try {
-        this.client
-          .acquireConnection()
-          .then((cnx) => {
-            cnx.__knexTxId = this.txid;
-            cnx.isTransaction = true;
-            resolve(cnx);
-          })
-          .catch(reject);
-      } catch (e) {
-        reject(e);
-      }
-    }).then(async (connection) => {
+
+    return this._acquireConnection(config).then(async (connection) => {
       try {
+        connection.__knexTxId = this.txid;
+        connection.isTransaction = true;
```

Comment:
Various sections of the `Transaction` code are manipulating values on the `connection` object.  This is problematic for a number of reasons.

In this specific case, I suspect that Oracle connections are not being released properly when a Transaction is more than 2 levels deep.